selfservice/selfserviceClient.go

```diff
@@ -45,6 +49,36 @@ func (sc *SelfServiceClient) queryServer(url string) []byte {
 	return responseBody
 }
 
+func (sc *SelfServiceClient) queryServerPost(url string, body []byte) []byte {
```

Comment:
I am not a big fan of the name `queryServerPost`.
We have another function called `queryServer` which performs GET.
Semantically then, this function should be called `sendToServer` or something along those lines.

We may have to rename both functions to achieve a good name -- and I am okay with that

Reply:
maybe one general function would be enough?

Reply:
That could also work. Up to you :)